explorer/client/src/components/ownedobjects/views/OwnedCoinView.tsx

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Coin } from '@mysten/sui.js';
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { ReactComponent as ContentIcon } from '../../../assets/SVGIcons/closed-content.svg';
+import { handleCoinType } from '../../../utils/stringUtils';
+import Longtext from '../../longtext/Longtext';
+import Pagination from '../../pagination/Pagination';
+import { type DataType, ITEMS_PER_PAGE } from '../OwnedObjectConstants';
+
+import styles from '../styles/OwnedCoin.module.css';
+
+function SingleCoinView({
+    results,
+    coinLabel,
+    currentPage,
+}: {
+    results: DataType;
+    coinLabel: string;
+    currentPage: number;
+}) {
+    const [isOpen, setIsOpen] = useState(false);
+
+    const switchIsOpen = useCallback(
+        () => setIsOpen((prevOpen) => !prevOpen),
+        []
+    );
+
+    // Switching the page closes any open group:
+    useEffect(() => {
+        setIsOpen(false);
+    }, [currentPage]);
+
+    const subObjList = results.filter(({ Type }) => Type === coinLabel);
+
+    return (
+        <div className={isOpen ? styles.openedgroup : styles.closedgroup}>
+            <div onClick={switchIsOpen} className={styles.summary}>
+                <div className={isOpen ? styles.openicon : styles.closedicon}>
+                    <ContentIcon />
+                </div>
+                <div>{handleCoinType(coinLabel)}</div>
+                <div>{subObjList.length}</div>
+                <div>
+                    {subObjList[0]._isCoin &&
+                    subObjList.every((el) => el.balance !== undefined)
+                        ? `${subObjList.reduce(
+                              (prev, current) => prev.add(current.balance!),
+                              Coin.getZero()
+                          )}`
+                        : ''}
+                </div>
+                <div />
```

Comment:
Empty div?

Reply:
This is a neat trick to help with the particular layout requirements.